CMakeLists.txt

```diff
@@ -1,140 +1,79 @@
-#
-# godot-cpp cmake arguments
-# GODOT_GDEXTENSION_DIR:		Path to the directory containing GDExtension interface header and API JSON file
-# GODOT_CPP_SYSTEM_HEADERS		Mark the header files as SYSTEM. This may be useful to supress warnings in projects including this one.
-# GODOT_CPP_WARNING_AS_ERROR	Treat any warnings as errors
-# GODOT_CUSTOM_API_FILE:		Path to a custom GDExtension API JSON file (takes precedence over `gdextension_dir`)
-# FLOAT_PRECISION:				Floating-point precision level ("single", "double")
-#
-# Android cmake arguments
-# CMAKE_TOOLCHAIN_FILE:		The path to the android cmake toolchain ($ANDROID_NDK/build/cmake/android.toolchain.cmake)
-# ANDROID_NDK:				The path to the android ndk root folder
-# ANDROID_TOOLCHAIN_NAME:	The android toolchain (arm-linux-androideabi-4.9 or aarch64-linux-android-4.9 or x86-4.9 or x86_64-4.9)
-# ANDROID_PLATFORM:			The android platform version (android-23)
-# More info here: https://godot.readthedocs.io/en/latest/development/compiling/compiling_for_android.html
+# CMAKE_BUILD_TYPE:			Compilation target (editor, template_debug, template_release)
+#
+# PLATFORM:	Platform type (LINUX, MACOS, WINDOWS, ANDROID, IOS, WEB)
+# Auto-detected by default depending on current OS or chosen toolchain
+#
+# other global and chosen platform-specific options:
+#
+# cmake -LH
+#
+# Note: use -Bbuild_dir option to build in separate directories
+# for different configurations
+#
+# cmake -Bbuild && cmake --build build
 #
 # Examples
 #
-# Builds a debug version:
+# Builds default configuration:
 # cmake .
 # cmake --build .
 #
-# Builds a release version with clang
-# CC=/usr/bin/clang CXX=/usr/bin/clang++ cmake -DCMAKE_BUILD_TYPE=Release -G "Unix Makefiles" .
+# Builds template_release version
+#
+# cmake -DCMAKE_BUILD_TYPE=template_release .
 # cmake --build .
 #
+# Creates multi-config setup and builds 2 library versions (example for Linux)
+#
+# cmake -G "Ninja Multi-Config" .
+# cmake --build . --config editor
+# cmake --build . --config template_release
+#
+# Builds web version, using Emscripten toolchain
+#
+# cmake --toolchain /usr/lib/emscripten/cmake/Modules/Platform/Emscripten.cmake .
+# cmake --build .
+#
+#
 # Builds an android armeabi-v7a debug version:
-# cmake -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake -DANDROID_NDK=$ANDROID_NDK \
-#		-DANDROID_TOOLCHAIN_NAME=arm-linux-androideabi-4.9 -DANDROID_PLATFORM=android-23 -DCMAKE_BUILD_TYPE=Debug .
+# cmake --toolchain $ANDROID_NDK/build/cmake/android.toolchain.cmake \
+# 	-DANDROID_TOOLCHAIN_NAME=arm-linux-androideabi-4.9 -DANDROID_TOOLCHAIN=clang -DANDROID_PLATFORM=21 .
 # cmake --build .
 #
-# Protip
-# Generate the buildfiles in a sub directory to not clutter the root directory with build files:
-# mkdir build && cd build && cmake -G "Unix Makefiles" .. && cmake --build .
 #
 # Todo
 # Test build for Windows, Mac and mingw.
 
-cmake_minimum_required(VERSION 3.12)
+cmake_minimum_required(VERSION 3.24)
```

Comment:
What are the changes that requite 3.24? We should use the lowest possible version to maintain stability and support, people shouldn't be forced to update unless some specific feature is required

Reply:
Indeed, many people work with LTS Linux distributions that only package older CMake versions (or they rely on the CMake version installed by Visual Studio).

I suggest not requiring a version newer than 3.16, since [that's what Ubuntu 20.04 has](https://packages.ubuntu.com/search?keywords=cmake).

Reply:
I'll revert to old versions in later updates

Reply:
> What are the changes that requite 3.24? We should use the lowest possible version to maintain stability and support, people shouldn't be forced to update unless some specific feature is required

it was made to support single `COMPILE_WARNING_AS_ERROR ON` , but i already have version issues on VS 2019, so i'll revert it.